### PR TITLE
Uses global id to construct lock key.

### PR DESCRIPTION
**Pull request by justinlittman**
refs #1087

## Why was this change made?
To eliminate one possibility for why locks are not being deleted from Redis, viz., objects are changing between a lock being created and the lock being deleted, resulting in the lock key also changing and therefore not being found to delete.

Note that this is at least one logical possibility for the problem. We won't know for sure until this is actually run in prod.

### CODE REVIEW

File: app/jobs/application_job.rb [hunk 1]
```diff
@@ -11,17 +11,24 @@
 # @see [resque-lock] https://github.com/defunkt/resque-lock/blob/master/lib/resque/plugins/lock.rb
 # @see [redis setnx] http://redis.io/commands/setnx
 class ApplicationJob < ActiveJob::Base
-  extend Resque::Plugins::Lock
+  include Resque::Plugins::Lock
```
Comment (jcoyne) [inline app/jobs/application_job.rb]: why did you change to `include`? The documentation says to `extend` https://github.com/defunkt/resque-lock#resque-lock

Comment (justinlittman) [reply app/jobs/application_job.rb]: Based on my (perhaps limited understanding), extends added the functions as class methods. As such, they ignored an override of the lock function (L29), which was also suggested by https://github.com/defunkt/resque-lock#resque-lock. It seems that include is the proper way to mixin the Lock module (unless it's not).
